### PR TITLE
Pelorus Operator v0.0.2

### PR DESCRIPTION
**Pull request by mpryc**
Pelorus Operator that addresses #777

To test:

`operator-sdk run bundle quay.io/pelorus/pelorus-operator-bundle:v0.0.2 --namespace pelorus`

Signed-off-by: Michal Pryc <mpryc@redhat.com>

@redhat-cop/mdt

### CODE REVIEW

File: scripts/pelorus-operator-patches/08_watches_dependent_resources.diff [hunk 1]
```diff
@@ -0,0 +1,9 @@
+--- watches.yaml.original	2023-01-18 17:44:32.236167254 +0100
++++ watches.yaml	2023-01-18 17:44:48.578216943 +0100
+@@ -3,4 +3,5 @@
+   version: v1alpha1
+   kind: Pelorus
+   chart: helm-charts/pelorus
++  watchDependentResources: false
```
Comment (weshayutin) [inline scripts/pelorus-operator-patches/08_watches_dependent_resources.diff:7]: nice

**Comment by mateusoliveira43** [comment]
If this is not in production yet (I checked in my cluster and latest is still 0.0.1) should we address #772 too? My fear is that the docs links can still change again

**Comment by mpryc** [comment]
/hold

some issues while installing this operator

**Comment by mpryc** [comment]
`
InstallErrorfailed to install release: rendered manifests contain a resource that already exists. Unable to continue with install: could not get information about the resource ClusterRoleBinding "pelorus-exporter-pelorus" in namespace "": clusterrolebindings.rbac.authorization.k8s.io "pelorus-exporter-pelorus" is forbidden: User "system:serviceaccount:pelorus:pelorus-operator-controller-manager" cannot get resource "clusterrolebindings" in API group "rbac.authorization.k8s.io" at the cluster scope
`

**Comment by mpryc** [comment]
note to myself:
  > Maybe we do want to list in the objects which are under `clusterPermissions` within our original rbac file created by the operator-sdk in the `watches.yaml`. My guess is that this file may be missing rbac auths because of the new flag.

**Comment by mpryc** [comment]
/unhold

The issues were because I've created new version without being logged in the cluster

Note: Add Instructions to generate new pelorus-operator in the docs.

**Comment by weshayutin** [comment]
```
(.venv) [whayutin@thinkdoe pelorus]$ operator-sdk run bundle quay.io/pelorus/pelorus-operator-bundle:v0.0.2 --namespace pelorus

INFO[0005] Creating a File-Based Catalog of the bundle "quay.io/pelorus/pelorus-operator-bundle:v0.0.2" 
INFO[0005] Generated a valid File-Based Catalog         
FATA[0005] Failed to run bundle: create catalog: error creating catalog source: namespaces "pelorus" not found 
(.venv) [whayutin@thinkdoe pelorus]$ oc create namespace pelorus
namespace/pelorus created
(.venv) [whayutin@thinkdoe pelorus]$ operator-sdk run bundle quay.io/pelorus/pelorus-operator-bundle:v0.0.2 --namespace pelorus
INFO[0004] Creating a File-Based Catalog of the bundle "quay.io/pelorus/pelorus-operator-bundle:v0.0.2" 
INFO[0005] Generated a valid File-Based Catalog         
INFO[0013] Created registry pod: quay-io-pelorus-pelorus-operator-bundle-v0-0-2 
INFO[0013] Created CatalogSource: pelorus-operator-catalog 
INFO[0014] OperatorGroup "operator-sdk-og" created      
INFO[0014] Created Subscription: pelorus-operator-v0-0-2-sub 
INFO[0017] Approved InstallPlan install-d4cr9 for the Subscription: pelorus-operator-v0-0-2-sub 
INFO[0017] Waiting for ClusterServiceVersion "pelorus/pelorus-operator.v0.0.2" to reach 'Succeeded' phase 
INFO[0017]   Waiting for ClusterServiceVersion "pelorus/pelorus-operator.v0.0.2" to appear 
INFO[0041]   Found ClusterServiceVersion "pelorus/pelorus-operator.v0.0.2" phase: Pending 
INFO[0044]   Found ClusterServiceVersion "pelorus/pelorus-operator.v0.0.2" phase: InstallReady 
INFO[0045]   Found ClusterServiceVersion "pelorus/pelorus-operator.v0.0.2" phase: Installing 
INFO[0065]   Found ClusterServiceVersion "pelorus/pelorus-operator.v0.0.2" phase: Succeeded 
INFO[0065] OLM has successfully installed "pelorus-operator.v0.0.2" 
(.venv) [whayutin@thinkdoe pelorus]$ oc get all -n pelorus
NAME                                                                  READY   STATUS      RESTARTS      AGE
pod/2ca9002371f6e33c34ac210e86f53e209de3a6504aee6d4c06978d4b64kssvr   0/1     Completed   0             78s
pod/committime-exporter-1-deploy                                      1/1     Running     0             13s
pod/committime-exporter-1-w67dj                                       0/1     Running     0             10s
pod/deploytime-exporter-1-d8gfk                                       0/1     Running     0             10s
pod/deploytime-exporter-1-deploy                                      1/1     Running     0             13s
pod/grafana-deployment-599878bd7d-xt578                               0/2     Init:0/1    0             3s
pod/grafana-deployment-777887458b-rhj9z                               0/2     Init:0/1    0             13s
pod/grafana-operator-controller-manager-7977d4bbd8-ccd25              2/2     Running     0             46s
pod/pelorus-operator-controller-manager-74957f5996-ss4bp              1/1     Running     0             50s
pod/prometheus-operator-7cc47f949d-4rpm4                              1/1     Running     0             52s
pod/prometheus-prometheus-pelorus-0                                   3/3     Running     1 (9s ago)    13s
pod/prometheus-prometheus-pelorus-1                                   3/3     Running     1 (10s ago)   13s
pod/quay-io-pelorus-pelorus-operator-bundle-v0-0-2                    1/1     Running     0             89s

NAME                                          DESIRED   CURRENT   READY   AGE
replicationcontroller/committime-exporter-1   1         1         0       13s
replicationcontroller/deploytime-exporter-1   1         1         0       13s

NAME                                                          TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)             AGE
service/committime-exporter                                   ClusterIP   172.30.105.253   <none>        8080/TCP            14s
service/deploytime-exporter                                   ClusterIP   172.30.22.167    <none>        8080/TCP            14s
service/grafana-operator-controller-manager-metrics-service   ClusterIP   172.30.108.125   <none>        8443/TCP            50s
service/grafana-service                                       ClusterIP   172.30.137.60    <none>        3000/TCP,9091/TCP   13s
service/pelorus-operator-controller-manager-metrics-service   ClusterIP   172.30.106.30    <none>        8443/TCP            54s
service/prometheus-operated                                   ClusterIP   None             <none>        9090/TCP            13s
service/prometheus-pelorus                                    ClusterIP   172.30.96.210    <none>        9090/TCP,9091/TCP   14s

NAME                                                  READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/grafana-deployment                    0/1     1            0           13s
deployment.apps/grafana-operator-controller-manager   1/1     1            1           46s
deployment.apps/pelorus-operator-controller-manager   1/1     1            1           50s
deployment.apps/prometheus-operator                   1/1     1            1           53s

NAME                                                             DESIRED   CURRENT   READY   AGE
replicaset.apps/grafana-deployment-599878bd7d                    1         1         0       3s
replicaset.apps/grafana-deployment-777887458b                    1         1         0       13s
replicaset.apps/grafana-operator-controller-manager-7977d4bbd8   1         1         1       46s
replicaset.apps/pelorus-operator-controller-manager-74957f5996   1         1         1       50s
replicaset.apps/prometheus-operator-7cc47f949d                   1         1         1       52s

NAME                                             READY   AGE
statefulset.apps/prometheus-prometheus-pelorus   2/2     13s

NAME                                                                        COMPLETIONS   DURATION   AGE
job.batch/2ca9002371f6e33c34ac210e86f53e209de3a6504aee6d4c06978d4b64819aa   1/1           10s        78s

NAME                                                     REVISION   DESIRED   CURRENT   TRIGGERED BY
deploymentconfig.apps.openshift.io/committime-exporter   1          1         1         config,image(committime-exporter:stable)
deploymentconfig.apps.openshift.io/deploytime-exporter   1          1         1         config,image(deploytime-exporter:stable)

NAME                                                 IMAGE REPOSITORY                                                               TAGS     UPDATED
imagestream.image.openshift.io/committime-exporter   image-registry.openshift-image-registry.svc:5000/pelorus/committime-exporter   stable   14 seconds ago
imagestream.image.openshift.io/deploytime-exporter   image-registry.openshift-image-registry.svc:5000/pelorus/deploytime-exporter   stable   14 seconds ago

NAME                                           HOST/PORT                                                                            PATH   SERVICES              PORT            TERMINATION          WILDCARD
route.route.openshift.io/committime-exporter   committime-exporter-pelorus.apps.cluster-wdh01032023a.wdh01032023a.mg.dog8code.com          committime-exporter   http                                 None
route.route.openshift.io/deploytime-exporter   deploytime-exporter-pelorus.apps.cluster-wdh01032023a.wdh01032023a.mg.dog8code.com          deploytime-exporter   http                                 None
route.route.openshift.io/grafana-route         grafana-route-pelorus.apps.cluster-wdh01032023a.wdh01032023a.mg.dog8code.com         /      grafana-service       grafana-proxy   reencrypt            None
route.route.openshift.io/prometheus-pelorus    prometheus-pelorus-pelorus.apps.cluster-wdh01032023a.wdh01032023a.mg.dog8code.com           prometheus-pelorus    web-tls         reencrypt/Redirect   None
(.venv) [whayutin@thinkdoe pelorus]$ 
```

**Comment by mpryc** [comment]
@weshayutin u happy with the deployment ? If yes could you pls merge?

**Pull request merged**